src/bun.js/node/types.zig

```diff
@@ -1835,548 +1849,2876 @@ pub const Emitter = struct {
+                        buf[bufSize - 2] != CHAR_DOT
+                    ) {
+                        if (bufSize > 2) {
+                            const lastSlashIndex = std.mem.lastIndexOfScalar(T, buf[0 ..bufSize], separator);
+                            if (lastSlashIndex == null) {
+                                res = "";
+                                bufSize = 0;
+                                lastSegmentLength = 0;
+                            } else {
+                                bufSize = lastSlashIndex.?;
+                                res = buf[0 ..bufSize];
+                                // Translated from the following JS code:
+                                //   lastSegmentLength =
+                                //     res.length - 1 - StringPrototypeLastIndexOf(res, separator);
+                                const lastIndexOfSep = std.mem.lastIndexOfScalar(T, buf[0 ..bufSize], separator);
+                                if (lastIndexOfSep == null) {
+                                    // Yes (>ლ), Node relies on the -1 result of 
+                                    // StringPrototypeLastIndexOf(res, separator).
+                                    // A - -1 is a positive 1.
+                                    // So the code becomes
+                                    //   lastSegmentLength = res.length - 1 + 1;
+                                    // or
+                                    //   lastSegmentLength = res.length;
+                                    lastSegmentLength = bufSize;
+                                } else {
+                                    lastSegmentLength = bufSize - 1 - lastIndexOfSep.?;
+                                }
+                            }
+                            lastSlash = i;
+                            dots = 0;
+                            continue;
+                        } else if (bufSize != 0) {
+                            res = "";
+                            bufSize = 0;
+                            lastSegmentLength = 0;
+                            lastSlash = i;
+                            dots = 0;
+                            continue;
+                        }
+                    }
+                    if (allowAboveRoot) {
+                        // Translated from the following JS code:
+                        //   res += res.length > 0 ? `${separator}..` : '..';
+                        if (bufSize > 0) {
+                            bufOffset = bufSize;
+                            bufSize += 1;
+                            buf[bufOffset] = separator;
+                            bufOffset = bufSize;
+                            bufSize += 2;
+                            buf[bufOffset] = CHAR_DOT;
+                            buf[bufOffset + 1] = CHAR_DOT;
+                        } else {
+                            bufSize = 2;
+                            buf[0] = CHAR_DOT;
```

Comment:
is this meant to be `bufOffset`

Reply:
Here it's supposed to be `bufSize` as it flows out of that `if` to below `res = buf[0 ..bufSize]`